instrumentation/opentelemetry-instrumentation-aiohttp-server/src/opentelemetry/instrumentation/aiohttp_server/__init__.py

```diff
@@ -0,0 +1,253 @@
+    )
+    if http_host_value_list:
+        result[SpanAttributes.HTTP_SERVER_NAME] = ",".join(
+            http_host_value_list
+        )
+    http_user_agent = request.headers.get("user-agent")
+    if http_user_agent:
+        result[SpanAttributes.HTTP_USER_AGENT] = http_user_agent
+
+    # remove None values
+    result = {k: v for k, v in result.items() if v is not None}
+
+    return result
+
+
+def set_status_code(span, status_code: int) -> None:
+    """Adds HTTP response attributes to span using the status_code argument."""
+
+    try:
+        status_code = int(status_code)
+    except ValueError:
+        span.set_status(
+            Status(
+                StatusCode.ERROR,
+                "Non-integer HTTP status: " + repr(status_code),
+            )
+        )
+    else:
+        span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, status_code)
+        span.set_status(
+            Status(http_status_to_status_code(status_code, server_span=True))
+        )
+
+
+class AiohttpGetter(Getter):
+    """Extract current trace from headers"""
+
+    def get(self, carrier, key: str) -> Union[List, None]:
+        """Getter implementation to retrieve an HTTP header value from the ASGI
+        scope.
+
+        Args:
+            carrier: ASGI scope object
+            key: header name in scope
+        Returns:
+            A list of all header values matching the key, or None if the key
+            does not match any header.
+        """
+        headers: CIMultiDictProxy = carrier.headers
+        if not headers:
+            return None
+        return headers.getall(key, None)
+
+    def keys(self, carrier: Dict) -> List:
+        return list(carrier.keys())
+
+
+getter = AiohttpGetter()
+
+
+@web.middleware
+async def middleware(request, handler):
+    """Middleware for aiohttp implementing tracing logic"""
+    if (
+        context.get_value("suppress_instrumentation")
+        or context.get_value(_SUPPRESS_HTTP_INSTRUMENTATION_KEY)
+        or _excluded_urls.url_disabled(request.url.path)
+    ):
+        return await handler(request)
+
+    span_name, additional_attributes = get_default_span_details(request)
+
+    req_attrs = collect_request_attributes(request)
+    duration_attrs = _parse_duration_attrs(req_attrs)
+    active_requests_count_attrs = _parse_active_request_count_attrs(req_attrs)
+
+    duration_histogram = meter.create_histogram(
+        name=MetricInstruments.HTTP_SERVER_DURATION,
+        unit="ms",
+        description="measures the duration of the inbound HTTP request",
+    )
+
+    active_requests_counter = meter.create_up_down_counter(
+        name=MetricInstruments.HTTP_SERVER_ACTIVE_REQUESTS,
+        unit="requests",
+        description="measures the number of concurrent HTTP requests those are currently in flight",
+    )
+
+    with tracer.start_as_current_span(
+        span_name,
+        kind=trace.SpanKind.SERVER,
+    ) as span:
```

Comment:
```suggestion
    with tracer.start_as_current_span(
        span_name,
        context=extract(request, getter=getter),
        kind=trace.SpanKind.SERVER,
    ) as span:
```

Without the correct `context`, this span will be orphaned.
```
from opentelemetry.propagate import extract
```